### PR TITLE
chore: prettier action

### PR DESCRIPTION
**Pull request by chandumlg**
## What are the changes introduced in this PR?

Introduces prettier github action. The action will fail if there are any unprettified files. Commit the prettified files and see the action go green.

## What is the related Linear task?

Resolves [INT-1238](https://linear.app/rudderstack/issue/INT-1238/standard-formatting-in-all-repositories)

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

File: .github/workflows/verify.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4.1.1
+        with:
+          # Make sure the actual branch is checked out when running on pull requests
+          ref: ${{ github.head_ref }}
+
+      - name: Prettify code
+        uses: creyD/prettier_action@v4.3
```
Comment (shrouti1507) [inline .github/workflows/verify.yaml:19]: @saikumarrs I remember we had a pre-commit hook as well right?

Comment (saikumarrs) [reply .github/workflows/verify.yaml:19]: @chandumlg, it's better we configure this action to run for only the changed files.
Additionally, we can run a scheduled cleanup action to create baseline.

Comment (chandumlg) [reply .github/workflows/verify.yaml:19]: Yeah. Either run only for changed filed or fix all and then run on all.

Comment (sanpj2292) [reply .github/workflows/verify.yaml:19]: Want to confirm one thing. This only checks if my source-code is not formatted along the lines that prettier has defined. Right ?!
What if there are lint failures & they somehow got committed ? They wouldn't be considered in this action right ?!

Comment (saikumarrs) [reply .github/workflows/verify.yaml:19]: Yeah, we need another action for it. However, we can only flag but not automatically fix them.

Comment (sanpj2292) [reply .github/workflows/verify.yaml:19]: Actually there is a way in which you can run only one command(eslint) with rules from eslint & prettier(ofcourse by disabling eslint-formatter rules) & flag if you find some changes. I tried it in one of our repositories. It is working well

Comment (saikumarrs) [reply .github/workflows/verify.yaml:19]: Does it also update the files with corrected formatting? If yes, great!

Comment (sanpj2292) [reply .github/workflows/verify.yaml:19]: Yes. It did

Comment (chandumlg) [reply .github/workflows/verify.yaml:19]: Closing this pr then @sanpj2292 . Let's have eslint do the formatting and linting.

**Pull request closed without merging**